addons/account/models/account_tax.py

```diff
@@ -267,6 +299,26 @@ def _check_company_consistency(self):
         if self._cr.fetchone():
             raise UserError(_("You can't change the company of your tax since there are some journal items linked to it."))
 
+    def _sanitize_vals(self, vals):
+        sanitized = vals.copy()
+        for doc_type in ('invoice', 'refund'):
+            fname = f"{doc_type}_repartition_line_ids"
+            if fname in sanitized:
+                repartition = sanitized.setdefault('repartition_line_ids', [])
+                repartition.extend([
+                    (command, id, {'document_type': doc_type, **v}) if command in (Command.CREATE, Command.UPDATE)
+                    else (command, id, v)
+                    for command, id, v in sanitized.pop(fname)
+                ])
+        return sanitized
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        return super().create([self._sanitize_vals(vals) for vals in vals_list])
+
+    def write(self, vals):
+        return super().write(self._sanitize_vals(vals))
```

Comment:
Because I don't know how to do that with a stored o2m field, then I would also need to sync pre-existing/new lines with `repartition_line_ids`. I feel like it is easier to use the ORM commands instead.